content/en/blog/2023/sunsetting-opencensus.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Sunsetting OpenCensus
+linkTitle: Sunsetting OpenCensus
+date: 2023-05-01
+author: '[Aaron Abbott](https://github.com/aabmass) (Google)'
+---
+
+In 2019, we announced that OpenTracing and OpenCensus would be merging to form
+the OpenTelemetry project. From the start, we considered OpenTelemetry
+[to be the next major version of both OpenTracing and OpenCensus](https://www.cncf.io/blog/2019/05/21/a-brief-history-of-opentelemetry-so-far/).
+
+We are excited to announce that OpenTelemetry has reached feature parity with
```

Comment:
Doesn't OpenCensus have stable logs?

Reply:
If you go to OTel-Python repo the logs API and SDK files both have underscores below them. In the OTel-Spec, the Logs API (if it ever comes into being) is not stable. Just a having a stable data model doesn't guarantee customers won't be broken in the future. We should be up front that there is a caveat about this "feature parity" claim. Customers could take a bet on OpenCensus Logging, whereas I'm not seeing that same confidence with OTel.

Reply:
OpenCensus does not include any logging API whatsoever.  It was just "stats(metrics) + traces".

Reply:
If OTel would make the same decision and call it done, then we'd be at feature parity. IMO, we should make customers aware that the logging pillar could break them in the future.

Reply:
@mattmccleary maybe I'm misunderstanding your point, but if OC only supported metrics and traces then doesn't feature parity with OTel only rely on metrics and traces? I am confused how the addition of logging in OTel isn't compatible with OC

Reply:
Exactly.  OpenCensus users don't need to migrate OC logging because there was nothing to migrate, it didn't exist.  We're asking OC users to migrate their metrics + traces to OpenTelemetry.

I don't understand the point you're making at all.

Reply:
Azure customers have taken a dependency on an Azure-specific OpenCensus Logs Exporter:
https://github.com/census-instrumentation/opencensus-python#log-exporter

If they migrate to OTLP Exporter for Logging, are you guaranteeing they will not be broken in the future?

Reply:
I see.  OpenCensus extensions are NOT part of this transition plan, nor will we be transitioning existing OpenCensus exporters.  The module you list would be considered both.

For the extension you mention, I'd suggest Azure could provide such a module to an OpenTelemetry Contrib component.  From what I can tell, it's very Azure specific (and does not even require the user to interact with OpenCensus to install it).  Additionally, you could still recommend Azure customers use that log exporter while they transition Metrics/Traces to OpenTelemetry via the OpenCesnus shim and I bet nothing breaks.

There are other extensions/contrib in opencensus, e.g. https://github.com/census-instrumentation/opencensus-java/tree/master/contrib/log_correlation/log4j2.   These will not break when using the OpenCensus-shim.

Reply:
> OpenCensus extensions are NOT part of this transition plan, nor will we be transitioning existing OpenCensus exporters.

Awesome. As long as this is clear to customers, great. It wasn't to me. Many of the customers I work with are not deeply versed in the communities like we are. They do not understand the difference between extensions vs core components. It's all OpenCensus to them.